File: 卡码/代码随想录/1/06-二叉树/Java/binaryTree/src/main/java/com/kurumi/binarytree/_02_二叉树属性/LeetCode_0222.java
//222. 完全二叉树的节点个数
package com.kurumi.binarytree._02_二叉树属性;

import com.kurumi.binarytree.TreeNode;

public class LeetCode_0222 {

}
class Solution {
  public int countNodes(TreeNode root) {
    if(root==null) return 0;
    TreeNode left=root.left;
    TreeNode right=root.right;
    int leftDepth=0,rightDepth=0;
    while(left!=null){
      left=left.left;
      leftDepth++;
    }
    while(right!=null){
      right=right.right;
      rightDepth++;
    }
    if(leftDepth==rightDepth) return (2<<leftDepth)-1;
    return 1+countNodes(root.left)+countNodes(root.right);
  }
}

File: 卡码/代码随想录/Go/二叉树/02-二叉树属性/LeetCode_0222.go
//222. 完全二叉树的节点个数
package main

import (
	. "binaryTree/types"
)

func countNodes(root *TreeNode) int {
	if root==nil {
		return 0
	}
	var left,right=root.Left,root.Right
	var leftDepth,rightDepth=0,0
	for left!=nil {
		left=left.Left
		leftDepth++
	}
	for right!=nil {
		right=right.Right
		rightDepth++
	}
	if leftDepth==rightDepth {
		return (2<<leftDepth)-1
	}
	return 1+countNodes(root.Left)+countNodes(root.Right)
}
